bin/devc.dart

```diff
@@ -20,31 +21,40 @@ void _showUsageAndExit() {
   exit(1);
 }
 
-void main(List<String> args) {
+CompilerOptions validateOptions(List<String> args) {
   var options = parseOptions(args);
-  if (options.help) _showUsageAndExit();
-
-  var srcOpts = options.sourceOptions;
-  if (!srcOpts.useMockSdk && srcOpts.dartSdkPath == null) {
-    print('Could not automatically find dart sdk path.');
-    print('Please pass in explicitly: --dart-sdk <path>');
-    exit(1);
-  }
-
-  if (options.inputs.length == 0) {
-    print('Expected filename.');
-    _showUsageAndExit();
+  if (!options.help) {
+    var srcOpts = options.sourceOptions;
+    if (!srcOpts.useMockSdk && srcOpts.dartSdkPath == null) {
+      print('Could not automatically find dart sdk path.');
+      print('Please pass in explicitly: --dart-sdk <path>');
+      exit(1);
+    }
+    if (options.inputs.length == 0) {
+      print('Expected filename.');
+      return null;
+    }
   }
+  return options;
+}
 
+Future<bool> compile(CompilerOptions options) async {
```

Comment:
In this case there's a hidden conversion, BatchCompiler.run()'s bool return is promoted to a future thanks to this async: added a line of comment (moved to `lib/src/compiler.dart`)